vault/seal/gcpckms/gcpckms.go

```diff
@@ -141,16 +141,17 @@ func (s *GCPCKMSSeal) SetConfig(config map[string]string) (map[string]string, er
 			return nil, errwrap.Wrapf("error initializing GCP CKMS seal client: {{err}}", err)
 		}
 
-		// Make sure cryto key exists in GCP
-		keyInfo, err := kmsClient.Projects.Locations.KeyRings.CryptoKeys.Get(s.parentName).Do()
-		if err != nil {
-			return nil, errwrap.Wrapf("error fetching GCP CKMS seal key information: {{err}}", err)
-		}
-		if keyInfo == nil {
-			return nil, errors.New("no key information returned")
+		// Make sure user has permissions to encrypt (also checks if key exists)
+		if _, err := s.client.Projects.Locations.KeyRings.CryptoKeys.
+			Encrypt(s.parentName, &cloudkms.EncryptRequest{
+				Plaintext: base64.StdEncoding.EncodeToString([]byte("vault-gcpckms-test")),
+			}).Do(); err != nil {
+			return nil, errwrap.Wrapf("failed to encryot with GCP CKMS - ensure the "+
+				"key exists and the service account has at least "+
+				"roles/cloudkms.cryptoKeyEncrypterDecrypter permission: {{err}", err)
 		}
-		s.currentKeyID.Store(keyInfo.Name)
 
+		s.currentKeyID.Store(s.parentName)
```

Comment:
Actually, shouldn't this be the Name from the response object like a regular Encrypt call below?  I know it is a tiny bit of extra encryption (but only at startup) but why don't you use the Encrypt call from the interface instead of duplicating the call here.

Reply:
The `Name` is the same as `Parent` in this case. The name is the FQDN of the key.

Reply:
Updated to use built-in encrypt.